### PR TITLE
Add [SpellInfo] and new method for [Player]

### PR DESCRIPTION
**Pull request by iThorgrim**
- Add new object [SpellInfo].
- Add player:GetSpells() for getting all learned spell.
- Add [Global] GetSpellInfo(spell_entry) for getting SpelInfo with spell_entry.

### CODE REVIEW

**Comment by Terrorblade** [comment]
Looks fine, but you should at least NYI returns for the unsupported emulators

**Comment by Foereaper** [comment]
Like mentioned in DM's, when it comes to things like new objects we need to make sure that we can have an equivalent on the other emulators to prevent API differences getting out of hand. I think @Niam5 is probably the appropriate Mangos resource to let us know if this is possible or not, or whether something needs changing to achieve 1:1.

File: TrinityCore/SpellInfoMethods.h [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+     * <pre>
+     * 0 : SpellAttr0
+     * 1 : SpellAttr1
+     * 2 : SpellAttr2
+     * 3 : SpellAttr3
+     * 4 : SpellAttr4
+     * 5 : SpellAttr5
+     * 6 : SpellAttr6
+     * 7 : SpellAttr7
+     * 8 : SpellCustomAttributes
```
[the remark this answers is not in the document]
Comment (iThorgrim) [reply TrinityCore/SpellInfoMethods.h]: It's done :)

**Comment by Foereaper** [comment]
We'll work on this PR and make sure it works with the other cores in the SpellInfoPR branch before merging into master

File: TrinityCore/SpellInfoMethods.h [hunk 2]
```diff
@@ -0,0 +1,922 @@
...
+        return 1;
+    }
+    
+    /**
+     * Checks if the [SpellInfo] has a specific aura.
+     *
+     * An aura represents a status change or modification due to a spell or ability. 
+     * These auras are identified by a predefined set of constants represented by the 'AuraType' enumeration.
+     *
+     * @param aura : The specific aura to check.
```
Comment (robinsch) [inline TrinityCore/SpellInfoMethods.h:227]: breaks doc, missing data type

File: TrinityCore/SpellInfoMethods.h [hunk 3]
```diff
@@ -0,0 +1,922 @@
...
+     * 3 : SpellAttr3
+     * 4 : SpellAttr4
+     * 5 : SpellAttr5
+     * 6 : SpellAttr6
+     * 7 : SpellAttr7
+     * -1 : SpellCustomAttributes
+     * </pre>
+     *
+     * @param int8 attributeType : the type of the attribute.
+     * @param attribute : the specific attribute to check.
```
Comment (robinsch) [inline TrinityCore/SpellInfoMethods.h:62]: breaks doc

File: TrinityCore/SpellInfoMethods.h [hunk 4]
```diff
@@ -0,0 +1,922 @@
...
+        return 1;
+    }
+    
+    /**
+     * Checks if the [SpellInfo] has a specific effect.
+     *
+     * A spell can have various effects such as damage, healing, or status changes.
+     * These effects are identified by a predefined set of constants represented by the 'SpellEffects' enumeration.
+     *
+     * @param effect : The specific effect to check.
```
Comment (robinsch) [inline TrinityCore/SpellInfoMethods.h:211]: breaks doc

Comment (iThorgrim) [reply TrinityCore/SpellInfoMethods.h]: Fix 🦾

File: TrinityCore/SpellInfoMethods.h (hunk 3 above)
Comment (iThorgrim) [reply TrinityCore/SpellInfoMethods.h:62]: Fix 🦾

File: TrinityCore/SpellInfoMethods.h (hunk 2 above)
Comment (iThorgrim) [reply TrinityCore/SpellInfoMethods.h]: Fix 🦾

**Comment by iThorgrim** [comment]
I'm taking the liberty of closing this PR for the time being, as the Eluna structure is not up to date on this branch, so I'll have to update it manually.

**Pull request closed without merging**